### PR TITLE
Eme deferred session closing

### PR DESCRIPTION
**Pull request by aczs**

### CODE REVIEW

File: media/server/main/source/MediaKeysServerInternal.cpp [hunk 1]
```diff
@@ -587,6 +594,61 @@ bool MediaKeysServerInternal::isNetflixKeySystemInternal(int32_t keySessionId) c
...
+
+    m_mainThread->enqueueTaskAndWait(m_mainThreadClientId, task);
+}
+
+void MediaKeysServerInternal::incrementSessionIdUsageCounterInternal(int32_t keySessionId)
+{
+    auto sessionIter = m_mediaKeySessions.find(keySessionId);
+    if (sessionIter == m_mediaKeySessions.end())
+    {
+        RIALTO_SERVER_LOG_ERROR("Failed to find the session");
```
Comment (skypetts) [inline media/server/main/source/MediaKeysServerInternal.cpp:613]: minor: log key session id?

File: media/server/main/source/MediaKeysServerInternal.cpp [hunk 2]
```diff
@@ -587,6 +594,61 @@ bool MediaKeysServerInternal::isNetflixKeySystemInternal(int32_t keySessionId) c
...
+
+    m_mainThread->enqueueTaskAndWait(m_mainThreadClientId, task);
+}
+
+void MediaKeysServerInternal::decrementSessionIdUsageCounterInternal(int32_t keySessionId)
+{
+    auto sessionIter = m_mediaKeySessions.find(keySessionId);
+    if (sessionIter == m_mediaKeySessions.end())
+    {
+        RIALTO_SERVER_LOG_ERROR("Failed to find the session");
```
Comment (skypetts) [inline media/server/main/source/MediaKeysServerInternal.cpp:633]: minor: log key session id?

File: media/server/main/source/MediaKeysServerInternal.cpp [hunk 3]
```diff
@@ -587,6 +594,61 @@ bool MediaKeysServerInternal::isNetflixKeySystemInternal(int32_t keySessionId) c
...
+        sessionIter->second.bufCounter--;
+    }
+
+    if (sessionIter->second.bufCounter == 0 && sessionIter->second.shouldBeDestroyed)
+    {
+        RIALTO_SERVER_LOG_INFO("Deferred closing of mksId %d", keySessionId);
+        MediaKeyErrorStatus status = sessionIter->second.mediaKeySession->closeKeySession();
+        if (MediaKeyErrorStatus::OK != status)
+        {
+            RIALTO_SERVER_LOG_ERROR("Failed to close the key session");
```
Comment (skypetts) [inline media/server/main/source/MediaKeysServerInternal.cpp:648]: minor: log key session id?

**Review by skywojciechowskim — APPROVED** [review]

**Review by skypetts — COMMENTED** [review]
Looks good to me but I think it need GS experts to approve

**Review by skywilliamsonl — APPROVED** [review]

**Review by skywojciechowskim — APPROVED** [review]

**Review by skypetts — APPROVED** [review]

**Pull request merged**